libsolidity/analysis/ImmutableValidator.h

```diff
@@ -29,7 +29,7 @@ namespace solidity::frontend
 /**
  * Validates access and initialization of immutable variables:
  * must be directly initialized in their respective c'tor
- * can not be read by any function/modifier called by the c'tor (or the c'tor itself)
+ * can not be read before being initialized or by another immutable variable
```

Comment:
wording is weird (variable reads variable) and also I'm not sure about the correctness either. Are you talking about the point where immutable state variables are initialized inline?